### PR TITLE
start with editing post

### PR DESCRIPTION
**Pull request by YuruBeDeniz**
---
name: Pull Request Template
about: Template for creating new pull requests
---

## Summary

I started working with edit button, edit post card is shown when the edit button is clicked and user can cancel the event if cancel button is clicked. However, getting the post with axios.get(`/api/post/${id}`) with useEffect() is not working. Hence, axios.put(`/api/post/${id}`, requestBody) request is not working either. Haven't figured it out yet, I will continue working on it later.

## Related Issue

edit routes are not working.

## Type of Change

EditPost.jsx is added, 2 routes to post.route.js is added.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no new functionality, just code cleanup/refactoring)
- [ ] Documentation update

## Screenshots (if applicable)

image

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)

### CODE REVIEW

File: server/routes/post.routes.js [hunk 1]
```diff
@@ -26,5 +26,24 @@ router.post("/", isAuthenticated, (req, res) => {
...
+  const postId = req.params.id;
+  Post.findById(postId)
+    .then(post => {
+      res.status(200).json(post);
+    })
+    .catch(err => console.log(err));
+})
+
+
+router.put("/id", (req, res) => {
```
Comment (antoniosaleme) [inline server/routes/post.routes.js:39]: I think, it should be:

router.put("/:id", (req, res) => {

File: server/routes/post.routes.js [hunk 2]
```diff
@@ -26,5 +26,24 @@ router.post("/", isAuthenticated, (req, res) => {
       })
 });
 
+router.get("/id", (req, res) => {
```
Comment (antoniosaleme) [inline server/routes/post.routes.js:29]: I think, it should be:

router.put("/:id", (req, res) => {

**Review by antoniosaleme — CHANGES_REQUESTED** [review]

File: client/src/components/Blog/EditPostCard.jsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    //console.log("check if updated: ", post);
+  };
+
+  useEffect(() => {
+    axios.get(`/api/post/${id}`)
+      .then(response => {
+        const { post } = response.data;
+        setPost(post);
+      })
+  }, [])
```
Comment (antoniosaleme) [inline client/src/components/Blog/EditPostCard.jsx:33]: I think you don't neet this one, you are already doing that on handleSubmit 🤔

File: server/routes/post.routes.js [hunk 4]
```diff
@@ -26,5 +26,24 @@ router.post("/", isAuthenticated, (req, res) => {
...
+      res.status(200).json(post);
+    })
+    .catch(err => console.log(err));
+})
+
+
+router.put("/id", (req, res) => {
+  console.log(req.params);
+  const { post } = req.body;
+  Post.findByIdAndUpdate(req.params.postId, { post }, { new: true })
```
Comment (antoniosaleme) [inline server/routes/post.routes.js:42]: this one, should be something like this:
const { title, body } = req.body;
  Post.findByIdAndUpdate(req.params.id, { title, body }, { new: true })

**Review by antoniosaleme — CHANGES_REQUESTED** [review]

File: client/src/components/Blog/EditPostCard.jsx [hunk 5]
```diff
@@ -0,0 +1,72 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+
+
+export default function EditPostCard({ post: initialPost, onCancel }) {
+  const [post, setPost] = useState(initialPost);
+  //console.log("POST: ", post);
+
+  const id = post._id;
```
Comment (antoniosaleme) [inline client/src/components/Blog/EditPostCard.jsx:9]: where are you using this one? 👀

**Review by antoniosaleme — CHANGES_REQUESTED** [review]

File: server/routes/post.routes.js [hunk 6]
```diff
@@ -26,5 +26,24 @@ router.post("/", isAuthenticated, (req, res) => {
       })
 });
 
+router.get("/id", (req, res) => {
+  const postId = req.params.id;
+  Post.findById(postId)
+    .then(post => {
```
Comment (antoniosaleme) [inline server/routes/post.routes.js:32]: don't forget to populate:

 .populate('author')

**Review by antoniosaleme — CHANGES_REQUESTED** [review]

**Review by Pablo-Mdz — APPROVED** [review]

**Pull request merged**